src/main/java/org/commonwl/view/workflow/WorkflowService.java

```diff
@@ -175,6 +176,13 @@ public Workflow getWorkflow(GitDetails gitInfo) {
                 // Add the new workflow if it exists
                 try {
                     createQueuedWorkflow(workflow.getRetrievedFrom());
+
+                    // Add the old commit for the purposes of permalinks
+                    // TODO: Separate concept of commit from branch ref, see #164
+                    GitDetails byOldCommitId = workflow.getRetrievedFrom();
+                    byOldCommitId.setBranch(workflow.getLastCommit());
```

Comment:
Does this still work OK if a `Workflow` already exists with that commit explicitly set?